File structure updates (#123)
OP-TEE/optee_docs: architecture/file_structure.rst

    * - include/crypto/
      - Architecture specific include files relating to /core/crypto or
        /core/arch/arm/crypto files

    * - include/kernel/
      - Architecture specific include files relating to /core/kernel or
        /core/arch/arm/kernel files

    * - include/mm/
      - Architecture specific include files relating to /core/mm or
        /core/arch/arm/mm files

    * - include/sm/
      - Include files relating to the secure monitor

    * - include/tee/
      - Architecture specific include files relating to /core/tee or
        /core/arch/arm/tee

    * - kernel/
      - Miscellaneous low level architecure specific files

    * - plat-\*/
      - Specific files for the different supported platform

    * - mm/
      - Memory management

    * - tee/
      - TEE files

    * - sm/
      - Secure Monitor, ARMv7-A only

lib/
*************
.. list-table::
    :header-rows: 1
    :widths: 1 5

    * - Directory
      - Description

    * - libdl/
      - Implementation of dlopen(), dlsym() and dlclose() used by TAs and ldelf

    * - libdl/include/
      - Include files for libdl

    * - libmbedtls/

>       - Mbedtls crypto library

Reviewer: ```suggestion
      - Mbed TLS crypto library
```

    * - libmbedtls/core/
      - Glue code only compiled with core to connect with the core internal
        <crypto/crypto.h> API.

    * - libmbedtls/include/

>       - Include files with configuration of mbedtls

Reviewer: ```suggestion
      - Include files with configuration of Mbed TLS
```

    * - libmbedtls/mbedtls/

>       - Top directory of the imported mbedtls source tree

Reviewer: ```suggestion
      - Top directory of the imported Mbed TLS source tree
```

>       - Top directory of the imported mbedtls source tree

Reviewer: ```suggestion
      - Top directory of the imported Mbed TLS source tree
```

    * - libmbedtls/mbedtls/include/

>       - Mbedtls include files

Reviewer: ```suggestion
      - Mbed TLS include files
```

    * - libmbedtls/mbedtls/library/

>       - Mbedtls implementation

Reviewer: ```suggestion
      - Mbed TLS implementation
```

    * - libunw/
      - Unwind library 

    * - libunw/include/
      - Include files for libunwnd

    * - libutee/
      - Libutee which provide the implementation of TEE Internal Core API.

    * - libutee/arch/

>       - Architecure specific implementation

Reviewer: ```suggestion
      - Architecture specific implementation
```

    * - libutee/include/
      - Include files relating to libutee and the header files for
        TEE Internal Core API

    * - libutils/
      - The reduced "libc" of OP-TEE

    * - libutils/ext/
      - Extensions to a standard libc

    * - libutils/ext/arch/
      - Architecture specific implmementation of the extensions

    * - libutils/ext/include/
      - Include files relating to the extensions

    * - libutils/isoc/
      - A subset of ISOC

    * - libutils/isoc/arch/
      - Architecture specific

    * - libutils/isoc/include/

>       - Header files relating to the provided subset of ISOC

Reviewer: ```suggestion
      - Header files related to the provided subset of ISOC
```

    * - libutils/isoc/newlib/
      - Routines imported from newlib

ta/
*************
.. list-table::
    :header-rows: 1
    :widths: 1 5

    * - Directory
      - Description

    * - trusted_keys
      - Trusted key TA

    * - trusted_keys/include
      - Header file of the ABI provided by the trusted key TA

    * - arch
      - Architecture specific files needed to compile a TA

    * - mk
      - Makefile includes needed to build TAs and the TA dev kit

    * - avb

>       - TA to support AVB (Android Verified Boot)

Reviewer: Worth pointing out that it's AVB 2.0 that we adhere to?

Author: This might age better without such details.

Reviewer: Sure

    * - avb/include
      - Header file of the ABI provided by the AVB TA

    * - pkcs11
      - TA to support PKCS#11 

    * - pkcs11/src

>       - Source code of the pkcs11 TA

Reviewer: ```suggestion
      - Source code for the PKCS#11 TA
```

    * - pkcs11/include

>       - Header file of the ABI provided by the pkcs11 TA

Reviewer: ```suggestion
      - Header file for the ABI provided by the PKCS#11 TA
```
